src/cpp/astar.cpp

```diff
@@ -27,15 +27,34 @@ bool operator<(const Node &n1, const Node &n2) {
 // See for various grid heuristics:
 // http://theory.stanford.edu/~amitp/GameProgramming/Heuristics.html#S7
 // L_\inf norm (diagonal distance)
-inline float linf_norm(int i0, int j0, int i1, int j1) {
-  return std::max(std::abs(i0 - i1), std::abs(j0 - j1));
+inline float linf_norm(int current_x, int current_y, int goal_x, int goal_y, int max_x, int max_y) {
```

Comment:
My C++ is a little rusty but I think you can indicate that arguments are unused by not naming them.

Reply:
Will check

Reply:
Seems your right